Makefile

```diff
@@ -1,25 +1,13 @@
 GOOS ?= $(shell go env GOOS)
 GOARCH ?= $(shell go env GOARCH)
-LDFLAGS='$(shell hack/version.sh)'
+VERSION ?= '$(shell hack/version.sh)'
 
 # Images management
 REGISTRY?="docker.io/karmada"
 REGISTRY_USER_NAME?=""
 REGISTRY_PASSWORD?=""
 REGISTRY_SERVER_ADDRESS?=""
 
-# Set your version by env or using latest tags from git
-VERSION?=""
-ifeq ($(VERSION), "")
-    LATEST_TAG=$(shell git describe --tags)
```

Comment:
binary version and image tag should be kept the same.
For image tag, it comes here.
we just need to add a `--dirty` flag `LATEST_TAG=$(shell git describe --tags --dirty)`?

Reply:
As you say `binary version and image tag should be kept the same.`  It's best way to use the same code. So I refactor it to get from `util::get_version` function.

If you keep the two ways to get version. It's easy to make different results, just like this bug.